lib/compat/wordpress-6.7/class-wp-block-templates-registry.php

```diff
@@ -0,0 +1,269 @@
+				$template->is_custom   = ! isset( $default_template_types[ $template_name ] );
+				$template->post_types  = isset( $args['post_types'] ) ? $args['post_types'] : array();
+			}
+
+			$this->registered_templates[ $template_name ] = $template;
+
+			return $template;
+		}
+
+		/**
+		 * Retrieves all registered templates.
+		 *
+		 * @since 6.7.0
+		 *
+		 * @return WP_Block_Template[]|false Associative array of `$template_name => $template` pairs.
+		 */
+		public function get_all_registered() {
+			return $this->registered_templates;
+		}
+
+		/**
+		 * Retrieves a registered template by its name.
+		 *
+		 * @since 6.7.0
+		 *
+		 * @param string $template_name Template name including namespace.
+		 * @return WP_Block_Template|null|false The registered template, or null if it is not registered.
+		 */
+		public function get_registered( $template_name ) {
+			if ( ! $this->is_registered( $template_name ) ) {
+				return null;
+			}
+
+			return $this->registered_templates[ $template_name ];
+		}
+
+		/**
+		 * Retrieves a registered template by its slug.
+		 *
+		 * @since 6.7.0
+		 *
+		 * @param string $template_slug Slug of the template.
+		 * @return WP_Block_Template|null The registered template, or null if it is not registered.
+		 */
+		public function get_by_slug( $template_slug ) {
+			$all_templates = $this->get_all_registered();
+
+			if ( ! $all_templates ) {
+				return null;
+			}
+
+			foreach ( $all_templates as $template ) {
+				if ( $template->slug === $template_slug ) {
+					return $template;
+				}
+			}
+
+			return null;
+		}
+
+		/**
+		 * Retrieves registered templates matching a query.
+		 *
+		 * @since 6.7.0
+		 *
+		 * @param array  $query {
+		 *     Arguments to retrieve templates. Optional, empty by default.
+		 *
+		 *     @type string[] $slug__in     List of slugs to include.
+		 *     @type string[] $slug__not_in List of slugs to skip.
+		 *     @type string   $post_type    Post type to get the templates for.
+		 * }
+		 */
+		public function get_by_query( $query = array() ) {
+			$all_templates = $this->get_all_registered();
+
+			if ( ! $all_templates ) {
+				return array();
+			}
+
+			$query            = wp_parse_args(
+				$query,
+				array(
+					'slug__in'     => array(),
+					'slug__not_in' => array(),
+					'post_type'    => '',
+				)
+			);
+			$slugs_to_include = $query['slug__in'];
+			$slugs_to_skip    = $query['slug__not_in'];
+			$post_type        = $query['post_type'];
+
+			$matching_templates = array();
+			foreach ( $all_templates as $template_name => $template ) {
+				if ( ! empty( $slugs_to_include ) && ! in_array( $template->slug, $slugs_to_include, true ) ) {
+					continue;
+				}
+
+				if ( ! empty( $slugs_to_skip ) && in_array( $template->slug, $slugs_to_skip, true ) ) {
+					continue;
+				}
+
+				if ( ! empty( $post_types ) && ! in_array( $post_type, $template->post_types, true ) ) {
+					continue;
+				}
+
+				$matching_templates[ $template_name ] = $template;
+			}
```

Comment:
```suggestion
			foreach ( $all_templates as $template_name => $template ) {
				if ( $slugs_to_include && ! in_array( $template->slug, $slugs_to_include, true ) ) {
					continue;
				}

				if ( $slugs_to_skip && in_array( $template->slug, $slugs_to_skip, true ) ) {
					continue;
				}

				if ( $post_types && ! in_array( $post_type, $template->post_types, true ) ) {
					continue;
				}

				$matching_templates[ $template_name ] = $template;
			}
```

We can make the expression simpler, since we don't actually need the functionality from `empty`.

Reply:
Merged, thanks! :+1: